src/types/action.ts

```diff
@@ -0,0 +1,64 @@
+import type {
+  ClientErrorStatusCode,
+  SuccessStatusCode,
+} from 'hono/utils/http-status'
+import type { TypedResponse } from './response.js'
+
+export type ActionResponse = {
+  /**
+   * HTTP response headers.
+   */
+  headers?: Record<string, string> | undefined
+  /**
+   * Message to show in the toast
+   *
+   * @example 'Action succeded!'
+   */
+  message: string
+  /**
+   * HTTP Status code to send the message with.
+   *
+   * Should match either 200 or 4xx status code.
+   *
+   * @example 200
+   */
+  statusCode: Extract<SuccessStatusCode, 200> | ClientErrorStatusCode
+}
+
+export type ActionResponseFn = (
+  response: ActionResponse,
+) => TypedResponse<ActionResponse>
+
+//////////////////////////////////////////////////////
+// Message Response
+
+export type MessageActionParameters = string
+export type MessageActionResponseFn = (
+  message: MessageActionParameters,
+) => TypedResponse<ActionResponse>
```

Comment:
Maybe we just rely on `ActionResponse` here.

Reply:
i will drop it as part of https://github.com/wevm/frog/pull/214#discussion_r1556465704

Reply:
refactored and deleted in 9a16f47